pandas/__init__.py

```diff
@@ -202,8 +202,40 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    # Check if old pyarrow is installed
+    from pandas.compat._optional import VERSIONS
+
+    try:
+        import pyarrow
+
+        pa_msg = (
+            f"was too old on your system - pyarrow {VERSIONS['pyarrow']} "
+            "is the current minimum supported version as of this release."
+        )
+        del pyarrow
+    except ImportError:
+        pa_msg = "was not found to be installed on your system."
+
+    warnings.warn(
+        f"""
+Pyarrow will become a required dependency of pandas in the next major release of pandas,
```

Comment:
```suggestion
Pyarrow will become a required dependency of pandas in the next major release of pandas (3.0),
```

I think we can be explicit about what this next major release will be

Reply:
> Is the next major release also the next minor release, i.e., 2.2 then 3.0? I noticed that the dev wheels recently switched to reporting `3.0.0.dev0` recently.

No, 2.2 is coming out sometime this week.
But yes, 3.0.0 will be the release after that.